### PR TITLE
refactor(explore): move MetricControl and FilterControl to sub-component

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

To make the explore code more maintainable, I will refactor some controls, so first I need to move some files to the sub-component.

- This change will not change any codebase logic
- This change will not change the filename

### TEST PLAN
IT / cypress run ok

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by ktmud — APPROVED** [review]
Good call!  Tested and everything still works as expected.

File: superset-frontend/spec/javascripts/explore/components/AdhocFilterOption_spec.jsx [hunk 1]
```diff
@@ -25,8 +25,8 @@ import Popover from 'src/common/components/Popover';
 import AdhocFilter, {
   EXPRESSION_TYPES,
   CLAUSES,
-} from 'src/explore/AdhocFilter';
-import AdhocFilterOption from 'src/explore/components/AdhocFilterOption';
+} from 'src/explore/components/controls/FilterControl/AdhocFilter';
+import AdhocFilterOption from '../../../../src/explore/components/controls/FilterControl/AdhocFilterOption';
```
Comment (villebro) [inline superset-frontend/spec/javascripts/explore/components/AdhocFilterOption_spec.jsx:29]:
```suggestion
import AdhocFilterOption from 'src/explore/components/controls/FilterControl/AdhocFilterOption';
```

File: superset-frontend/spec/javascripts/explore/components/AdhocFilterEditPopoverSqlTabContent_spec.jsx [hunk 2]
```diff
@@ -25,8 +25,8 @@ import { FormGroup } from 'react-bootstrap';
 import AdhocFilter, {
   EXPRESSION_TYPES,
   CLAUSES,
-} from 'src/explore/AdhocFilter';
-import AdhocFilterEditPopoverSqlTabContent from 'src/explore/components/AdhocFilterEditPopoverSqlTabContent';
+} from 'src/explore/components/controls/FilterControl/AdhocFilter';
+import AdhocFilterEditPopoverSqlTabContent from '../../../../src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSqlTabContent';
```
Comment (villebro) [inline superset-frontend/spec/javascripts/explore/components/AdhocFilterEditPopoverSqlTabContent_spec.jsx:29]:
```suggestion
import AdhocFilterEditPopoverSqlTabContent from 'src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSqlTabContent';
```

File: superset-frontend/spec/javascripts/explore/components/AdhocMetricEditPopoverTitle_spec.jsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ import sinon from 'sinon';
 import { shallow } from 'enzyme';
 import { Tooltip } from 'src/common/components/Tooltip';
 
-import AdhocMetricEditPopoverTitle from 'src/explore/components/AdhocMetricEditPopoverTitle';
+import AdhocMetricEditPopoverTitle from '../../../../src/explore/components/controls/MetricControl/AdhocMetricEditPopoverTitle';
```
Comment (villebro) [inline superset-frontend/spec/javascripts/explore/components/AdhocMetricEditPopoverTitle_spec.jsx:25]:
```suggestion
import AdhocMetricEditPopoverTitle from 'src/explore/components/controls/MetricControl/AdhocMetricEditPopoverTitle';
```

**Review by villebro — APPROVED** [review]
LGTM, just a few import nits

**Comment by zhaoyongjie** [comment]
> LGTM, just a few import nits

Ville, Thanks for the point out.

**Pull request merged**